### PR TITLE
ENH: to_list as alias for tolist

### LINKED ISSUE #8826: ENH: Add Series.to_list, alias for tolist
Hello,

IPython autocomplete (tab) shows

```
In [1]: ts.to
ts.to_clipboard  ts.to_frame      ts.to_period     ts.to_string
ts.to_csv        ts.to_hdf        ts.to_pickle     ts.to_timestamp
ts.to_dense      ts.to_json       ts.to_sparse     ts.tolist
ts.to_dict       ts.to_msgpack    ts.to_sql
```

maybe `tolist` should be named `to_list`
and `tolist` should be marked as deprecated
because it seems that other methods follow
an implicit naming convention (`to_...`) . 

Kind regards

### PR DESCRIPTION
**Pull request by retkowski**
- [x] closes #8826
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -554,6 +554,8 @@ def tolist(self):
         """
         return list(self)
 
+    to_list = tolist
```
Comment (jreback) [inline pandas/core/arrays/categorical.py]: can you update the tests to also test ``to_list`` (just 1 is enough)

File: doc/source/whatsnew/v0.24.0.txt [hunk 2]
```diff
@@ -911,6 +911,7 @@ Other API Changes
 - Slicing a single row of a DataFrame with multiple ExtensionArrays of the same type now preserves the dtype, rather than coercing to object (:issue:`22784`)
 - :class:`DateOffset` attribute `_cacheable` and method `_should_cache` have been removed (:issue:`23118`)
 - :meth:`Index.hasnans` and :meth:`Series.hasnans` now always return a python boolean. Previously, a python or a numpy boolean could be returned, depending on circumstances (:issue:`23294`).
+- ``Series.tolist()`` and ``Index.tolist()`` now have an alias ``to_list`` (:issue:`8826`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: can you use the ``:func:`Series.to_list` `` reference and so on; pls add to api.rst as well.

**Review by jreback — CHANGES_REQUESTED** [review]
I would not be completely averse to hiding ``.tolist`` in the tab completion. you can add to ``_dir_deletions`` to achieve this.

**Comment by ScientiaEtVeritas** [comment]
I've added a commit which reflects the points you mentioned.

File: doc/source/api.rst [hunk 3]
```diff
@@ -324,6 +324,7 @@ Conversion
    Series.to_period
    Series.to_timestamp
    Series.tolist
```
Comment (jreback) [inline doc/source/api.rst]: can you remove the tolist

File: pandas/tests/test_base.py [hunk 4]
```diff
@@ -1074,9 +1074,10 @@ class TestToIterable(object):
         'method',
         [
             lambda x: x.tolist(),
+            lambda x: x.to_list(),
             lambda x: list(x),
             lambda x: list(x.__iter__()),
```
Comment (jreback) [inline pandas/tests/test_base.py]: you will need to run the test and then catch any instance of .tolist as it will show a warning

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_base.py (hunk 4 above)
Comment (ScientiaEtVeritas) [reply pandas/tests/test_base.py]: Hm. I've run the tests (via ``pytest pandas/tests/test_base.py -v -k TestToIterable`` and I don't get any warnings there.

Comment (jreback) [reply pandas/tests/test_base.py]: pls run the test suite itself. likely this is used elsewhere which need catching.

File: pandas/core/arrays/categorical.py [hunk 5]
```diff
@@ -323,7 +323,7 @@ class Categorical(ExtensionArray, PandasObject):
     # ops, which raise
     __array_priority__ = 1000
     _dtype = CategoricalDtype(ordered=False)
-    _deprecations = frozenset(['labels'])
+    _deprecations = frozenset(['labels', 'tolist'])
```
Comment (jreback) [inline pandas/core/arrays/categorical.py]: add a note whereever you are doing this that tolist is not actually deprecated, just suppressed in the ``__dir__``. We *could* also make an attribute for this, but prob not worth it

File: pandas/tests/test_base.py (hunk 4 above)
Comment (ScientiaEtVeritas) [reply pandas/tests/test_base.py]: It seems that I can't produce any warning related to these changes. I also quick-checked the Travis logs and compared them with other builds.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @TomAugspurger @jorisvandenbossche if any comments.

**Comment by TomAugspurger** [comment]
We use `.tolist()` in the docs in a few places. If we're recommend to_list
then we need to update those.

Personally, I don't care which we recommend. NumPy users will be familiar
with tolist. But if we're remvoing
tolist from the API docs, then we should update all the docs.

**Comment by jreback** [comment]
can you merge master and update the docs as indicated.

**Review by jreback — CHANGES_REQUESTED** [review]
pls merge master and update as discussed in the comments

**Comment by ScientiaEtVeritas** [comment]
I think I've updated all places where ``tolist`` is used.

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.24.0.rst [hunk 6]
```diff
@@ -1071,6 +1071,7 @@ Other API Changes
 - :meth:`Index.hasnans` and :meth:`Series.hasnans` now always return a python boolean. Previously, a python or a numpy boolean could be returned, depending on circumstances (:issue:`23294`).
 - The order of the arguments of :func:`DataFrame.to_html` and :func:`DataFrame.to_string` is rearranged to be consistent with each other. (:issue:`23614`)
 - :meth:`CategoricalIndex.reindex` now raises a ``ValueError`` if the target index is non-unique and not equal to the current index. It previously only raised if the target index was not of a categorical dtype (:issue:`23963`).
+- :func:`Series.tolist` and :func:`Index.tolist()` now have an alias ``to_list`` (:issue:`8826`)
```
[the remark this answers is not in the document]
Comment (ScientiaEtVeritas) [reply doc/source/whatsnew/v0.24.0.rst]: Fixed

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ScientiaEtVeritas

**Pull request merged**